src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3967,19 +3967,10 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
             var hasArgList = this.CurrentToken.Kind == SyntaxKind.ArgListKeyword;
 
             TypeSyntax type = null;
-            if (!hasArgList)
-            {
-                type = this.ParseType(mode: ParseTypeMode.Parameter);
-            }
-            else if (this.IsPossibleType())
```

Comment:
this code branch is what we hit if 'hasArgList' is true.  However, if 'hasArgList' is true, then "IsPossibleType" will always be false.  That's because IsPossibleType checks if CurrentToken is a predefined-type, or is a 'true identifier' . "ArgListKeyword" is neither of those.